src/nerdle/solver.py

```diff
@@ -66,23 +66,90 @@ def get_best_guess(self, possible_solutions: Set[str], available_guesses: Set[st
 
 
 class DeeperMinimaxSolver(Solver):
+    N_BRANCH = 5
+
+    def __init__(self, scorer: Scorer) -> None:
+        self.minimax = MinimaxSolver(scorer)
+        super().__init__(scorer)
+
     def get_best_guess(self, possible_solutions: Set[str], available_guesses: Set[str]) -> str:
+        """Search one level deeper for guesses that have better next round worst case scenarios."""
 
-        best_guess_to_date = Guess("N/A", 1_000_000, 1_000_000)
+        if len(possible_solutions) == 1:
+            return possible_solutions.pop()
+
+        best_guess = Guess("N/A", 1_000_000, 1_000_000, {})
+        best_largest_bucket_to_date = 1_000_000
+
+        bucket_guesses = {}
 
         for guess in available_guesses:
             possible_solutions_by_score = self.get_possible_solutions_by_score(
                 possible_solutions, guess
             )
-
-            # TODO look at possible solutions for subsequent round of guesses
-
             new_guess = Guess.create(guess, possible_solutions_by_score)
+            b_size = new_guess.size_of_largest_bucket
 
-            if new_guess.improves_upon(best_guess_to_date, possible_solutions):
-                best_guess_to_date = new_guess
+            if b_size not in bucket_guesses:
+                bucket_guesses[b_size] = []
 
-        return best_guess_to_date.word
+            bucket_guesses[b_size].append(new_guess)
+
+        if 1 in bucket_guesses:
+            # if any of the guesses would get us there next go, just choose one
+            for guess in bucket_guesses[1]:
+                if guess.improves_upon(best_guess, possible_solutions):
+                    best_guess = guess
+        else:
+            # only search the most plausible guesses
+            search_guesses = []
+            total_count = 0
+            for b_size in sorted(bucket_guesses):
+                search_guesses.extend(bucket_guesses[b_size])
+                total_count += len(bucket_guesses[b_size])
+                if (b_size == 1) or (total_count >= self.N_BRANCH):
```

Comment:
if (b_size == 1) or len(search_guesses) >= self.N_BRANCH:

I think!

Reply:
I should have commented this more clearly.

I'm trying to keep adding potential plausible guesses until we have over self.N_BRANCH plausible guesses to try. I think your approach could end up with very large numbers of plausible guesses (e.g. if the best had 53 in its worst bucket and then there was one guess with 52, one with 51, one with 50 etc...)

Reply:
Might be worth implementing a greater than, less than and equality method for Guess so that we can just sort using the heuristic and cut the potential plausible guesses at some point the list sorted by 'goodness'

Reply:
I understand now - we don't need to track `total_count` because we can just use the length of the list we are building - agreed